src/GrpAb/Map.jl

```diff
@@ -339,12 +339,13 @@ end
 
 Returns whether $h$ is surjective.
 """
-function is_surjective(A::GrpAbFinGenMap)
+@attr Bool function is_surjective(A::GrpAbFinGenMap)
   if isfinite(codomain(A))
     H, mH = image(A)
     return (order(codomain(A)) == order(H))::Bool
   else
-    return (order(cokernel(A)[1]) == 1)::Bool
+    KK = cokernel(A)[1]
+    return (is_finite(KK) && order(KK) == 1)::Bool
```

Comment:
```suggestion
    return is_finite(KK) && isone(order(KK))
```

Reply:
I am also not sure if the conversion into `Bool` is strictly necessary. You probably know that better than I do.

Reply:
You are right. It is not necessary anymore